client/src/featureform/register.py

```diff
@@ -3947,6 +4010,194 @@ def search(self, raw_query, local=False):
             return search(processed_query, self._host)
 
 
+class ColumnResource:
+    """
+    Base class for all column resources. This class is not meant to be instantiated directly.
+    In the original syntax, features and labels were registered using the `register_resources`
+    method on the sources (e.g. SQL/DF transformation or tables sources); however, in the new
+    Class API syntax, features and labels can now be declared as class attributes on an entity
+    class. This means that all possible params for either resource must be passed into this base
+    class prior to calling `register_column_resources` on the registrar.
+    """
+    def __init__(
+        self,
+        transformation_args: tuple,
+        type: Union[ColumnTypes, str],
+        resource_type: str,
+        entity: Union[Entity, str],
+        variant: str,
+        owner: Union[str, UserRegistrar],
+        inference_store: Union[str, OnlineProvider, FileStoreProvider],
+        timestamp_column: str,
+        description: str,
+        schedule: str,
+        tags: List[str],
+        properties: Dict[str, str],
+    ):
+        registrar, source_name_variant, columns = transformation_args
+        self.type = type if isinstance(type, str) else type.value
+        self.registrar = registrar
+        self.source = source_name_variant
+        self.entity_column = columns[0]
+        self.source_column = columns[1]
+        self.resource_type = resource_type
+        self.entity = entity
+        self.variant = variant
+        self.owner = owner
+        self.inference_store = inference_store
+        self.timestamp_column = timestamp_column
+        self.description = description
+        self.schedule = schedule
+        self.tags = tags
+        self.properties = properties
+
+    def register(self):
+        features, labels = self.get_features_and_labels()
+
+        self.registrar.register_column_resources(
+            source=self.source,
+            entity=self.entity,
+            entity_column=self.entity_column,
+            owner=self.owner,
+            inference_store=self.inference_store,
+            features=features,
+            labels=labels,
+            timestamp_column=self.timestamp_column,
+            schedule=self.schedule,
+        )
+    
+    def get_features_and_labels(self) -> Tuple[List[ColumnMapping], List[ColumnMapping]]:
```

Comment:
would it make sense to separate these into two methods? If not would it make sense to drop the get_ in the name. Super minor nitpick feel free to ignore